architecture/core_concepts/builds_and_image_streams.adoc

```diff
@@ -175,134 +175,335 @@ endif::[]
+
+* If your image stream tag follows another image stream tag (it does not point directly to a docker image tag), it is possible that the image stream tag will be updated to follow a different image stream tag in the future. Again, this could result in picking up an incompatible version change.
+
+=== Important terms
+
+Docker repository::
+A collection of related docker images and tags identifying them.
+For example, the OpenShift Jenkins images are in a Docker repository:
++
+----
+docker.io/openshift/jenkins-2-centos7
+----
+
+Docker registry::
+A content server that can store and service images from Docker repositories. 
+For example:
++
+----
+registry.access.redhat.com
+----
+
+Docker image::
+ A specific set of content that can be run as a container. Usually associated with a particular tag within a Docker repository.
+
+Docker image tag::
+A label applied to a Docker image in a repository that distinguishes a specific image. For example, here *3.6.0* is a tag:
+
+----
+docker.io/openshift/jenkins-2-centos7:3.6.0
+----
+
+[NOTE]
+====
+A Docker image tag can be updated to point to new Docker image content at any time.
+====
+
+Docker image ID::
+A SHA (Secure Hash Algorithm) code that can be used to pull an image. For example:
+
+----
+docker.io/openshift/jenkins-2-centos7@sha256:ab312bda324
+----
+
+[NOTE]
+====
+A SHA image ID cannot change. A specific SHA identifier always references the exact same docker image content.
+====
+
+Image stream::
+An {product-title} object that contains pointers to any number of Docker-formatted container images identified by tags. You can think of an image stream
+as equivalent to a Docker repository.
 
+Image stream tag::
+A named pointer to an image in an image stream. An image stream tag is similar to a Docker image tag. See xref:image-stream-tag[Image Stream Tag] below.
+
+Image stream image::
+An image that allows you to retrieve a specific Docker image from a particular image stream where it is tagged. 
+An image stream image is an API resource object that pulls together some metadata about a particular image SHA identifier.
+See xref:image-stream-image[Image Stream Images] below.
+
+Image stream trigger::
+A trigger that causes a specific action when an image stream tag changes. For example, importing can cause the value of the tag to change, which 
+causes a trigger to fire when there are Deployments, Builds, or other resources listening for those. See xref:image-stream-triggers[Image Stream Triggers] below.
+
+[[image-stream-configure]]
+=== Configuring Image Streams
+
+ifdef::openshift-origin,openshift-online,openshift-enterprise,openshift-dedicated[]
```

Comment:
@mburke5678 - something is not right here. This ifdef doesn't have a closing endif.

Reply:
@vikram-redhat That ifdef was in the file when I started on this issue. I thought the same thing; but, assumed that someone knew better than I. Same with the following one. Thank you for adding the endif back.